### PR TITLE
Use package to format prices in Plan Store

### PR DESCRIPTION
**Pull request by alshakero**
#### Changes proposed in this Pull Request

* Instead of creating own formatting logic, make the plan store use the format-currency package to format prices. 

#### Testing instructions

1. Go to https://calypso.live/new?branch=update/plan-store-price-formats.
2. Make sure the plans grid looks good.

#### Note
I tested with #48963 locally and it works great. I (or Marco) will rebase #48963 once this is merged.

### CODE REVIEW

File: packages/data-stores/src/plans/resolvers.ts [hunk 1]
```diff
@@ -55,19 +43,17 @@ function getMonthlyPrice( plan: PricedAPIPlan ) {
...
-		return `${ price }${ currency.symbol }`;
-	}
-	// else
-	return `${ currency.symbol }${ price }`;
+/**
+ * Formats the plan price according to 'format-currency' package rules
+ * We use this for consistency in prices formats across monthly and annual plans
+ *
+ * @param plan the plan object
+ */
```
Comment (ciampo) [inline packages/data-stores/src/plans/resolvers.ts:54]: Nitpick: why are we using JSDoc notation in this file? Can we switch to TypeScript-only?

File: packages/data-stores/src/plans/test/resolvers.ts [hunk 2]
```diff
@@ -20,15 +20,15 @@ describe( 'getSupportedPlans', () => {
 				// and rounds to 0 decimal places
 				currency_code: 'INR',
 				product_slug: PLAN_FREE,
-				raw_price: 13,
+				raw_price: 0,
```
Comment (ciampo) [inline packages/data-stores/src/plans/test/resolvers.ts:21]: We should remove the mention of the decimal places from the comment (a few lines above), now that this comes from an external package.

Maybe the comment should change to something like "Free plan, with a cost of 0"

File: packages/data-stores/src/plans/test/resolvers.ts [hunk 3]
```diff
@@ -20,15 +20,15 @@ describe( 'getSupportedPlans', () => {
...
 				product_id: 1,
 			},
 			{
 				// This currency formats the symbol after the number
 				// and rounds to 0 decimal places
 				currency_code: 'INR',
 				product_slug: PLAN_PREMIUM,
-				raw_price: 13,
+				raw_price: 12,
```
Comment (ciampo) [inline packages/data-stores/src/plans/test/resolvers.ts:28]: We should remove the mention of the decimal places from the comment (a few lines above), now that this comes from an external package.

Maybe the comment should be changed to "Premium plan, billed annually"

File: packages/data-stores/src/plans/test/resolvers.ts [hunk 4]
```diff
@@ -38,6 +38,7 @@ describe( 'getSupportedPlans', () => {
 				product_slug: PLAN_PREMIUM_MONTHLY,
 				raw_price: 13,
 				product_id: 3,
+				bill_period: 31,
```
Comment (ciampo) [inline packages/data-stores/src/plans/test/resolvers.ts:37]: We should remove the mention of the decimal places from the comment (a few lines above), now that this comes from an external package.

Maybe the comment should be changed to "Premium plan, billed monthly"

**Review by ciampo — APPROVED** [review]
Tested in Gutenboarding, Focused Launch and Step by Step launch (with EUR as currency) — everything works as expected.

Just left few minor comments that won't affect the behaviour of the code — Once they're addressed, this PR is good to be merged

File: packages/data-stores/src/plans/resolvers.ts (hunk 1 above)
Comment (alshakero) [reply packages/data-stores/src/plans/resolvers.ts:54]: If you only add JSDoc to explain the function without listing the params, the linter overlord won't let you.

**Pull request merged**